docs/rules/no-await-sync-events.md

```diff
@@ -0,0 +1,53 @@
+# Disallow unnecessary `await` for sync events (no-await-sync-events)
+
+Ensure that sync events are not awaited unnecessarily.
+
+## Rule Details
+
+Functions in the event object provided by Testing Library, including
+fireEvent and userEvent, do NOT return Promise, with an exception of
+`userEvent.type`. Some examples are:
```

Comment:
This is technically true, but you don't need to always wait for it. From [user-event doc](https://github.com/testing-library/user-event#typeelement-text-options):

> To be clear, userEvent.type always returns a promise, but you only need to await the promise it returns if you're using the delay option. Otherwise everything runs synchronously and you can ignore the promise.

So should this rule report about `userEvent.type` if no delay option passed? I think so

Reply:
I agree with that ☝️  we should only await if the delay option is used. It would be great if we can cover that scenario with the rule

Reply:
Agree. 
the rule is, only if userEvent.type with delay option, await is valid. For all the other cases, it disallows await.